src/WebJobs.Script.Grpc/MessageExtensions/GrpcMessageConversionExtensions.cs

```diff
@@ -101,8 +101,25 @@ internal static TypedData ToModelBindingDataArray(this ParameterBindingData[] da
 
         internal static async Task<TypedData> ToRpcHttp(this HttpRequest request, ILogger logger, GrpcCapabilities capabilities)
         {
+            var handlesHttpRouteParamsWhenProxying = !string.IsNullOrEmpty(capabilities.GetCapabilityState(RpcWorkerConstants.HandlesRouteParamsWhenHttpProxying));
+            var skipHttpInputs = !string.IsNullOrEmpty(capabilities.GetCapabilityState(RpcWorkerConstants.HttpUri));
```

Comment:
Would these capabilities have to be rechecked for each HTTP request? Wondering if it's worth trying to optimize this if it may be a hot path.

Reply:
One possible optimization is to have a Capability type with boolean properties for each of these capabilities. This type should be initialized/updated during worker init/env reload response. Using that will bring a minor improvement as the dictionary look up can be removed. but that changes goes beyond the scope of this PR. We can consider making that change and update a bunch of places to use that.